g2p/mappings/tokenizer.py

```diff
@@ -67,7 +67,14 @@ def __init__(self, mapping: Mapping):
     def _build_regex(self):
         if not self.case_sensitive:
             self.inventory = [c.lower() for c in self.inventory]
+        # Remove the indices, they're not part of the text input for the rules
         self.inventory = [re.sub(r"{[0-9]+}", "", x) for x in self.inventory]
+        # Rules with "in": "è|é" have two distinct inputs, split them.
```

Comment:
This approach is still not completely right - for example, `^b` means `b` at the beginning of the string, but this code to assemble the inventory sees the caret as a literal instead. I'm not sure how to fix this problem globally correctly, but this patch specifically address alternations in rules, the way they are used in `mic` and `gla`.

The tokenization inventory is only important for non-letter symbols used as letters - e.g., the apostrophe in `mic`, because the tokenizer assume anything that is defined as a letter in the Unicode standard is part of the word. So, while not perfect nor fully general, this solutions fixes the actual use we have on hand at the moment.

Reply:
If ^ is a legitimate character in the orthography, then `escape_special` should be set to `true`, otherwise we could safely strip special regex characters and not treat them as part of the inventory right? the only case where we wouldn't strip is in alternations which you've handled here. So for ^ or $ or other things, if `escape_special` is false, we could just strip them out. What do you think of that @joanise ?

Reply:
That sounds like better logic, I agree. I'm still not fully satisfied with my approach, because we just know there will be more corner cases. This patch really only fixes the cases occurring in current mappings.

I think a much better solution would be to rewrite my tokenizer from scratch using a letter inventory for the language instead, but for now let's just merge this in.

I'll turn your suggestion for caret into an issue so we remember the idea, but it won't be a priority just now.